lib/components/joystick/joystick_action.dart

```diff
@@ -113,37 +114,17 @@ class JoystickAction {
   }
 
   void render(Canvas c) {
-    if (_rectBackgroundDirection != null && _dragging && enableDirection) {
-      if (spriteBackgroundDirection == null) {
-        final double radiusBackground = _rectBackgroundDirection.width / 2;
-        c.drawCircle(
-          Offset(
-            _rectBackgroundDirection.left + radiusBackground,
-            _rectBackgroundDirection.top + radiusBackground,
-          ),
-          radiusBackground,
-          _paintBackground,
-        );
-      } else {
-        spriteBackgroundDirection.renderRect(c, _rectBackgroundDirection);
-      }
```

Comment:
mostly extracted this bit to a method, it was repeated a lot. made it easier to update it to use render instead of renderRect.